src/realm/object-store/class.cpp

```diff
@@ -22,36 +22,29 @@
 
 namespace realm {
 
-Class::Class(const std::shared_ptr<Realm>& r, StringData object_type)
-    : m_realm(r)
-    , m_object_schema(&*r->schema().find(object_type))
-    , m_table(r->read_group().get_table(m_object_schema->table_key))
+Class::Class(std::shared_ptr<Realm> r, StringData object_type)
+    : m_realm(std::move(r))
+    , m_object_schema(&*m_realm->schema().find(object_type))
```

Comment:
Isn't this invalid if `object_type` isn't in the schema? Is that checked for?

(I know that is probably a programmer error, but it seems better to check and assert if this is part of core's public API)